code/obj/machinery/nuclearbomb.dm

```diff
@@ -215,12 +216,40 @@ ADMIN_INTERACT_PROCS(/obj/machinery/nuclearbomb, proc/arm, proc/set_time_left)
 		command_alert("\A [src] has been armed in [isturf(src.loc) ? get_area(src) : src.loc]. It will detonate in [src.get_countdown_timer()] minutes. All personnel must report to [get_area(src)] to disarm the bomb immediately.", "Nuclear Weapon Detected")
 		if (!ON_COOLDOWN(global, "nuke_planted", 20 SECONDS))
 			playsound_global(world, 'sound/machines/bomb_planted.ogg', 75)
+		if(src.record)
+			if(is_music_playing())
+				src.visible_message(SPAN_NOTICE("The [src]'s record player detects conflicting music and ejects the record!"))
+				src.record.set_loc(get_turf(src))
+				src.record = null
+			else
+				SPAWN(6 SECONDS) // Length of the "Bomb planted" sound effect
```

Comment:
The block under this needs to be indented for this to actually pause anything- otherwise it's just going to spawn a new 'thread' that waits 6 seconds to do nothing

Reply:
It seemed to work in testing, decreasing this number from 8>6 did make it match better with the nuke planted sound effect, it works as intended where the rest ran and the ghost message activated, but it waited to play the music.

Reply:
Increasing it to 20 also changed it, its working as intended where it only pauses playing the music and not the log or ghost message.